src/airflow_tools/providers/http_to_data_lake/operators/http_to_data_lake.py

```diff
@@ -176,6 +178,12 @@ def _file_name(self, n_part) -> str:
     def _response_filter(self, response) -> BytesIO:
         if not self.jmespath_expression:
             self.data = response.json()
+
+        elif (
+            self.jmespath_expression
+            and self.save_format in self.text_response_save_format
+        ):
+            self.data = response.text
```

Comment:
Shouldn't this be a value error? If we have a jmespath expression and the save_format is not json that's not a valid state. Also isn't self.jmespath_expression always truthy at this point? I think we're missing some cases, like if jmespath_expression is None and save_format isn't JSON, it will error when it does `response.json`

Reply:
You are right, I missed that case.

Reply:
I think I have solved this. The state of jmeshpath_expression do not matter for the text_responses, so the state of jmespath should not matter. If you want I can insert a raise Exception in the init if the save_format is on th elist of text_responses and jmespath_expression is not none, but I think it is not necessary.

Reply:
Theoretically it does not matter but in practice if a user of this library gives a jmespath_expression it means they don't understand what it does or have used the operator incorrectly and may not work like they expect. At the very least I would add a warning that it's being ignored.

Reply:
After your suggestions, I added an Exception